src/wallet/rpcwallet.cpp

```diff
@@ -4267,7 +4282,57 @@ UniValue z_viewtransaction(const UniValue& params, bool fHelp)
         addMemo(entry, memo);
         outputs.push_back(entry);
     }
-    // TODO unified addresses, orchard, see #5186
+
+    std::vector<uint256> ovksVector(ovks.begin(), ovks.end());
+    OrchardActions orchardActions = pwalletMain->GetOrchardWallet().GetTxActions(wtx, ovksVector);
+
+    // Orchard spends
+    for (auto & pair  : orchardActions.GetSpends()) {
+        // TODO test with Orchard spends after rebased on feature/wallet-orchard
+        auto actionIdx = pair.first;
+        OrchardActionSpend orchardActionSpend = pair.second;
+        auto outpoint = orchardActionSpend.GetOutPoint();
+        auto receivedAt = orchardActionSpend.GetReceivedAt();
+        auto noteValue = orchardActionSpend.GetNoteValue();
+        // if mapWallet doesn't contain the spent outpoint's txid,
+        // the wallet is corrupt, so using `.at` here is fine.
+
+        UniValue entry(UniValue::VOBJ);
+        entry.pushKV("type", ADDR_TYPE_ORCHARD);
+        entry.pushKV("spend", (int) actionIdx);
+        entry.pushKV("txidPrev", outpoint.hash.GetHex());
+        entry.pushKV("outputPrev", (int) outpoint.n);
+        auto ua = pwalletMain->FindUnifiedAddressByReceiver(receivedAt);
+        assert(ua.has_value());
+        std::string addrStr = keyIO.EncodePaymentAddress(ua.value());
+        entry.pushKV("address", addrStr);
+        entry.pushKV("value", ValueFromAmount(noteValue));
+        entry.pushKV("valueZat", noteValue);
+        spends.push_back(entry);
+    }
+
+    // Orchard outputs
+    for (const auto& [actionIdx, orchardActionOutput]  : orchardActions.GetOutputs()) {
+        auto noteValue = orchardActionOutput.GetNoteValue();
+        auto recipient = orchardActionOutput.GetRecipient();
+        auto memo = orchardActionOutput.GetMemo();
+
+        // Show the address that was cached at transaction construction as the
+        // recipient.
+        std::string address = keyIO.EncodePaymentAddress(
+            pwalletMain->GetPaymentAddressForRecipient(txid, recipient)
+        );
+
+        UniValue entry(UniValue::VOBJ);
+        entry.pushKV("type", ADDR_TYPE_ORCHARD);
+        entry.pushKV("output", (int) actionIdx);
```

Comment:
The help text calls this `actionoutput`, but again I think this should be `action` for consistency (since it's referring to an index into `vActionsOrchard`):
```suggestion
        entry.pushKV("action", (int) actionIdx);
```
Update the help text to be consistent.